### PR TITLE
DT-961 Initialise index with mapping and pipeline

### PR DESCRIPTION
**Pull request by andymarke**
TODO on create - wire to service and test behaviour of mapping and pipeline

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepository.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import uk.gov.justice.digital.hmpps.indexer.model.SyncIndex
+import uk.gov.justice.digital.hmpps.indexer.service.Offender
+
+const val pipelineId = "pnc-pipeline"
+
+@Repository
+class OffenderRepository {
+  @Qualifier("elasticSearchClient")
+  @Autowired
+  lateinit var highLevelClient: RestHighLevelClient
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepository.kt:23]: Can this be injected in the constructor?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepository.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  companion object {
+    val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun save(offender: Offender, index: SyncIndex) = log.info("would save an offender to {}", index)
+
+  fun createIndex(index: SyncIndex): Unit {
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepository.kt:31]: This seem a more flexible way of doing it.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        assertThatThrownBy { offenderRepository.createIndex(GREEN) }.hasMessageContaining("already exists")
+      }
+    }
+  }
+
+  @Nested
+  inner class Mappings {
+    lateinit var mappingProperties: Map<String, Any>
+    @BeforeEach
+    internal fun setUp() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt:87]: Just wondering why some tests are internal and others aren't.  Does it make a difference and which is better(er)?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+fun GetMappingsResponse.properties(index: String): Map<String, Any> {
+  val mapping = this.mappings()[index]?.sourceAsMap()!!
+  val mappingProperties = mapping["properties"] as Map<*, *>
+  @Suppress("UNCHECKED_CAST")
+  return mappingProperties as Map<String, Any>
+}
+
+fun Map<*, *>.value(property: String) : String {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt:187]: Nice!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepository.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepository.kt:31]: Though I think Unit is extraneous - it's the default return value. (equivalent to Java's void)

**Pull request merged**